js/foundation.magellan.js

```diff
@@ -109,18 +110,13 @@ class Magellan {
    * @function
    */
   scrollToLoc(loc) {
-    // Do nothing if target does not exist to prevent errors
-    if (!$(loc).length) {return false;}
     this._inTransition = true;
-    var _this = this,
-        scrollPos = Math.round($(loc).offset().top - this.options.threshold / 2 - this.options.barOffset);
-
-    $('html, body').stop(true).animate(
-      { scrollTop: scrollPos },
-      this.options.animationDuration,
-      this.options.animationEasing,
-      function() {_this._inTransition = false; _this._updateActive()}
-    );
+    var _this = this;
+
+    Foundation.SmoothScroll.scrollToLoc(loc, this.options, function() {
```

Comment:
I do not think it is a good idea of merging an object option to the SmoothScroll.defaults within this **static** method. The method doesn't have the reference to `this` instance. The reason I decided of making this method a **static** is **Magellan** module can just use it directly without creating an instance.

The merging of defaults and explicitly defined options are happened within Object instantiation. In this case in the constructor method.